presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -211,7 +221,14 @@ private static boolean isValidJsonObjectKeyType(Type type)
 
         static ObjectKeyProvider createObjectKeyProvider(Type type)
         {
-            String baseType = type.getTypeSignature().getBase();
+            TypeSignature signature = type.getTypeSignature();
+            String baseType = signature.getBase();
+            if (signature.isLongEnum()) {
```

Comment:
Maybe you can incorporate this into the `switch (baseType)` part now?

Reply:
how do you mean? inside the `default` case?

Reply:
Oh nvm, I thought now the baseType would have "enum" in it.